Sources/Xcodeproj/pbxproj().swift

```diff
@@ -330,6 +311,34 @@ public func pbxproj(xcodeprojPath: AbsolutePath, graph: PackageGraph, extraDirs:
             // Also add the source file to the compile phase.
             compilePhase.addBuildFile(fileRef: srcFileRef)
         }
+
+        // Add the `include` group for a libary C language target.
+        if case let clangModule as ClangModule = module, clangModule.type == .library, isDirectory(clangModule.includeDir) {
+            let includeDir = clangModule.includeDir
+            let includeGroup = makeGroup(for: includeDir)
+            for header in try walk(includeDir) where header.extension == "h" {
+                let group = makeGroup(for: header.parentDirectory)
+                group.addFileReference(path: header.basename)
+            }
+
+            // Generate a module map for ClangModule (if not provided by user) and
+            // add to the build settings.
+            targetSettings.common.DEFINES_MODULE = "YES"
+            let moduleMapPath: AbsolutePath
+            // If user provided the modulemap no need to generate.
+            if isFile(clangModule.moduleMapPath) {
+                moduleMapPath = clangModule.moduleMapPath
+            } else {
+                // Generate and drop the modulemap inside Xcodeproj folder.
+                let path = xcodeprojPath.appending(components: "GeneratedModuleMap", clangModule.c99name)
+                var moduleMapGenerator = ModuleMapGenerator(for: clangModule)
+                try moduleMapGenerator.generateModuleMap(inDir: path)
+                moduleMapPath = path.appending(component: moduleMapFilename)
+            }
+
+            includeGroup.addFileReference(path: moduleMapPath.asString, name: moduleMapPath.basename)
```

Comment:
Oh right — it will use the path if the name isn't provided, so what I mentioned only works in the "normal" case in which a file reference is group-relative and contains just the name as the path.